tools/check-metadata.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+import argparse
+import json
+import os
+import subprocess
+import sys
+from collections import OrderedDict
+
+
+ROOT_DIR = os.path.abspath(
+    os.path.join(os.path.dirname(os.path.realpath(__file__)), os.pardir))
+TOOLS_DIR = os.path.join(ROOT_DIR, "tools")
+DATA_DIR = os.path.join(ROOT_DIR, "data")
+SOUNDS_DIR = os.path.join(DATA_DIR, "sounds")
+
+SYSTEM_METADATA_PATH = os.path.join(DATA_DIR, "metadata.json")
+METADATA_SCHEMA_PATH = os.path.join(ROOT_DIR, "ci", "metadata.schema.json")
+
+
+def check_media_file_integrity(filepath):
+    stream = ffmpeg.input(filepath).output("pipe:", format="null")
+    try:
+        stream.run(capture_stdout=True, capture_stderr=True)
+    except ffmpeg.Error:
+        return False
+    return True
+
+
+def check_sorted(metadata):
+    if not sorted(metadata) == list(metadata):
+        return False
+
+    for sound in metadata:
+        props = list(metadata[sound])
+        if not sorted(props[1:]) == props[1:]:
+            return False
+
+
+def check_files(metadata, check_media_integrity=False):
+    valid = True
+    for sound in metadata:
+        location = metadata[sound].get("sound-file")
+        if not location:
+            locations = metadata[sound].get("sound-files")
+        else:
+            locations = [location]
+
+        for location in locations:
+            full_location = os.path.abspath(os.path.join(SOUNDS_DIR, location))
+            file_extension = os.path.splitext(full_location)[1]
+
+            if not os.path.isfile(full_location):
+                valid = False
+
+                print(f"File '{full_location}' does not exist.",
+                      file=sys.stderr)
+            elif (check_media_integrity and
+                    not check_media_file_integrity(full_location)):
+                valid = False
+                print(f"File '{full_location}' is not a valid media file.",
+                      file=sys.stderr)
+            if file_extension != ".webm":
+                print(f"File '{full_location}' is not a 'webm' file. "
+                      "Did you forget to convert it? "
+                      "Use ./tools/convert-sounds.sh", file=sys.stderr)
+    return valid
+
+
+def check_schema():
+    proccess = subprocess.Popen(
```

Comment:
A lot of this file consists of shelling out to other tools, like in this function. Maybe I am missing the context of the reasoning for this decision, but I'm not yet convinced that this is an improvement over the one line `jsonschema -i data/metadata.json ci/metadata.schema.json` that we had previously in the Travis config file.

Reply:
Yes, it does exactly what `jsonschema -i data/metadata.json ci/metadata.schema.json` does right now.
I developed this script with the idea of not only having a correct JSON file that follows our schema, but also to check that the indicated file paths exist and are valid, and also to warn you in the case you forget it to convert it to webm.
It would be bad to have a "sound-location" property pointing to a non-existing sound file.

Reply:
That could be done with a separate utility which is also called in the CI config file, or put all together in a local shell script which is called in the CI config file. I think if we have a Python utility which consists mostly of shelling out to other utilities, then it's a sign we are using the wrong tool for the job.